app/repositories/onlinetesting/OnlineTestEvaluationRepository.scala

```diff
@@ -0,0 +1,149 @@
+    }
+  }
+}
+
+class Phase1EvaluationMongoRepository()(implicit mongo: () => DB)
+  extends ReactiveRepository[ApplicationReadyForEvaluation, BSONObjectID]("application", mongo,
+    ApplicationReadyForEvaluation.applicationReadyForEvaluationFormats,
+    ReactiveMongoFormats.objectIdFormats) with OnlineTestEvaluationRepository[ApplicationReadyForEvaluation] {
+
+  val phase = PHASE1
+
+  val evaluationApplicationStatuses = List(ApplicationStatus.PHASE1_TESTS, ApplicationStatus.PHASE1_TESTS_PASSED, ApplicationStatus.PHASE2_TESTS)
+
+  def nextApplicationsReadyForEvaluation(currentPassmarkVersion: String, batchSize: Int): Future[List[ApplicationReadyForEvaluation]] = {
+    val query = BSONDocument("$and" -> BSONArray(
+      BSONDocument("applicationStatus" -> BSONDocument("$in" -> evaluationApplicationStatuses)),
+      BSONDocument(s"progress-status.${ProgressStatuses.PHASE1_TESTS_RESULTS_RECEIVED}" -> true),
+      BSONDocument("$or" -> BSONArray(
+        BSONDocument(s"testGroups.$phase.evaluation.passmarkVersion" -> BSONDocument("$exists" -> false)),
+        BSONDocument(s"testGroups.$phase.evaluation.passmarkVersion" -> BSONDocument("$ne" -> currentPassmarkVersion))
+      ))
+    ))
+
+    implicit val reader = bsonReader(doc => {
+      val bsonPhase1 = doc.getAs[BSONDocument]("testGroups").flatMap(_.getAs[BSONDocument](phase))
+      val phase1 = bsonPhase1.map(Phase1TestProfile.bsonHandler.read).get
+      applicationBSONReader(doc)(phase1.activeTests)(None)
+    })
+    selectRandom[ApplicationReadyForEvaluation](query, batchSize)
+  }
+}
+
+class Phase2EvaluationMongoRepository()(implicit mongo: () => DB)
+  extends ReactiveRepository[ApplicationReadyForEvaluation, BSONObjectID]("application", mongo,
+    ApplicationReadyForEvaluation.applicationReadyForEvaluationFormats,
+    ReactiveMongoFormats.objectIdFormats) with OnlineTestEvaluationRepository[ApplicationReadyForEvaluation] {
+
+  val phase = PHASE2
+
+  val prevPhase = PHASE1
+
+  val evaluationApplicationStatuses = List(ApplicationStatus.PHASE2_TESTS)
+
+  val evaluationProgressStatus = ProgressStatuses.PHASE2_TESTS_RESULTS_RECEIVED
+
+  def nextApplicationsReadyForEvaluation(currentPassmarkVersion: String, batchSize: Int): Future[List[ApplicationReadyForEvaluation]] = {
+    val query = BSONDocument("$and" -> BSONArray(
+      BSONDocument("applicationStatus" -> BSONDocument("$in" -> evaluationApplicationStatuses)),
+      BSONDocument(s"progress-status.${ProgressStatuses.PHASE2_TESTS_RESULTS_RECEIVED}" -> true),
+      BSONDocument(s"testGroups.$prevPhase.evaluation.passmarkVersion" -> BSONDocument("$exists" -> true)),
+      BSONDocument("$or" -> BSONArray(
+        BSONDocument(s"testGroups.$phase.evaluation.passmarkVersion" -> BSONDocument("$exists" -> false)),
+        BSONDocument(s"testGroups.$phase.evaluation.passmarkVersion" -> BSONDocument("$ne" -> currentPassmarkVersion)),
+        BSONDocument("$where" ->
```

Comment:
Hopefully mongo will do some optimisation instead of doing table scan as we got some additional filter criteria